pandas/tests/reshape/merge/test_multi.py

```diff
@@ -460,6 +460,33 @@ def test_merge_na_keys(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_merge_datetime_index_empty_df(self):
+
+        date = np.array(
+            [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
+            dtype=np.datetime64,
+        )
+        panel = np.array(["A", "B"], dtype=object)
+        data = np.array([1.5, 1.5], dtype=np.float64)
+
+        frame = DataFrame({"date": date, "panel": panel, "data": data}).set_index(
+            ["date", "panel"]
+        )
+        other = DataFrame(columns=["date", "panel", "state"]).set_index(
+            ["date", "panel"]
+        )
+
+        state = np.array([np.nan, np.nan], dtype=object)
+
+        expected = DataFrame(
```

Comment:
explicity construct the expected frame not using the inputs at all

Reply:
@PCerles can you do this